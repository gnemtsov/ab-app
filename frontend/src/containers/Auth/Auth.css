.Auth {
    margin: 20px auto;
    width: 80%;
    text-align: center;
    box-shadow: 0 2px 3px #ccc;
    border: 1px solid #eee;
    padding: 10px;
    box-sizing: border-box;
}

@media (min-width: 600px) {
    .Auth {
        width: 500px;
    }
}