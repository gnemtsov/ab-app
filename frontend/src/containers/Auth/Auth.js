import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

import Input from '../../components/UI/Input/Input';
import Button from '../../components/UI/Button/Button';
import Spinner from '../../components/UI/Spinner/Spinner';
import classes from './Auth.css';
import * as actionTypes from "../../store/actionTypes";
import { updateObject, checkValidity } from '../../shared/utility';

class Auth extends Component {
    state = {
        controls: {
            login: {
                elementType: 'input',
                elementConfig: {
                    type: 'text',
                    placeholder: 'Login'
                },
                value: 'admin',
                validation: {
                    required: true
                },
                valid: false,
                touched: false
            },
            password: {
                elementType: 'input',
                elementConfig: {
                    type: 'password',
                    placeholder: 'Password'
                },
                value: 'abapp',
                validation: {
                    required: true,
                    minLength: 6
                },
                valid: false,
                touched: false
            }
        }
    }

    inputChangedHandler = (event, controlName) => {
        const updatedControls = updateObject(this.state.controls, {
            [controlName]: updateObject(this.state.controls[controlName], {
                value: event.target.value,
                valid: checkValidity(event.target.value, this.state.controls[controlName].validation),
                touched: true
            })
        });
        this.setState({ controls: updatedControls });
    }

    submitHandler = (event) => {
        event.preventDefault();
        this.props.onAuth(this.state.controls.login.value, this.state.controls.password.value);
    }

    render() {
        const formElementsArray = [];
        for (let key in this.state.controls) {
            formElementsArray.push({
                id: key,
                config: this.state.controls[key]
            });
        }

        let form = formElementsArray.map(formElement => (
            <Input
                key={formElement.id}
                elementType={formElement.config.elementType}
                elementConfig={formElement.config.elementConfig}
                value={formElement.config.value}
                invalid={!formElement.config.valid}
                shouldValidate={formElement.config.validation}
                touched={formElement.config.touched}
                changed={(event) => this.inputChangedHandler(event, formElement.id)} />
        ));

        if (this.props.loading) {
            form = <Spinner />
        }

        let errorMessage = null;

        if (this.props.error) {
            errorMessage = (
                <p>{this.props.error.message}</p>
            );
        }

        let authRedirect = null;
        if (this.props.isAuthenticated) {
            authRedirect = <Redirect to="/" />
        }

        return (
            <div className={classes.Auth}>
                {authRedirect}
                {errorMessage}
                <form onSubmit={this.submitHandler}>
                    {form}
                    <Button btnType="Success">SUBMIT</Button>
                </form>
            </div>
        );
    }
}

const mapStateToProps = state => {
    return {
        loading: state.auth.loading,
        error: state.auth.error,
        isAuthenticated: state.auth.isAuthenticated
    };
};

const mapDispatchToProps = dispatch => {
    return {
        onAuth: (login, password) => dispatch({
            type: actionTypes.S_LOGIN,
            login: login,
            password: password
        })
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Auth);