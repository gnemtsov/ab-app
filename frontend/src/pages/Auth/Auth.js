import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

import Form from '../../containers/Form/Form';
import * as actionTypes from "../../store/actionTypes";

class Auth extends Component {
    render() {
        const authRedirect = this.props.isAuthenticated ? <Redirect to="/" /> : null;
        const conf = {
            buttonText: ['Login', 'Logging in..']
        }

        return (
            <div>
                {authRedirect}
                <Form
                    conf={conf}
                    title="Please, authenticate"
                    submitted={this.props.onAuth}
                    api="/auth/login" />
            </div>
        );
    }
}

const mapStateToProps = state => {
    return { isAuthenticated: state.auth.isAuthenticated };
};

const mapDispatchToProps = dispatch => {
    return {
        onAuth: (tokens) => dispatch({
            type: actionTypes.S_LOGIN,
            tokens: tokens
        })
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Auth);