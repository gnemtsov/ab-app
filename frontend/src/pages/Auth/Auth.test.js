import React from 'react';

import configureStore from 'redux-mock-store';

import { configure, render, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter } from "react-router-dom";

import Auth from './Auth';

configure({ adapter: new Adapter() });

describe('Auth', () => {
	it('Renders correctly when authenticated', () => {
		const mockStore = configureStore();
		const store = mockStore({
			auth: {
				isAuthenticated: true
			}
		}); 
		
		const wrapper = render(
			<BrowserRouter>
				<Auth store={store}/>
			</BrowserRouter>
		);
		expect(wrapper).toMatchSnapshot();
	});
	
	it('Renders correctly when not authenticated', () => {
		const mockStore = configureStore();
		const store = mockStore({
			auth: {
				isAuthenticated: false
			}
		}); 
		
		const wrapper = render(
			<BrowserRouter>
				<Auth store={store}/>
			</BrowserRouter>
		);
		expect(wrapper).toMatchSnapshot();
	});
})

