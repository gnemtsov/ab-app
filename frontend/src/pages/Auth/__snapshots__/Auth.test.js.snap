// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Renders correctly when authenticated 1`] = `
<div>
  <h1>
    Please, authenticate
  </h1>
  <div
    class="Loader"
  >
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
  </div>
</div>
`;

exports[`Auth Renders correctly when not authenticated 1`] = `
<div>
  <h1>
    Please, authenticate
  </h1>
  <div
    class="Loader"
  >
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
    <div>
      
    </div>
  </div>
</div>
`;
